### PR TITLE
fix: correctly handle tag and digest

### PR DESCRIPTION
**Pull request by rmutuleanu**
#### What does this PR do?

Fixes the case when the docker daemon is unavailable, and both the image digest and tag are present.

#### Where should the reviewer start?

The reviewer can start with the newly added unit test to better understand the issue.

#### Any background context you want to provide?

Snyk container scanning fails when the docker daemon is unavailable, and both the image tag and digest are passed. This is because of a bug in the code that extracts the image repository from the provided target image argument.

### CODE REVIEW

File: test/lib/analyzer/image-inspector.spec.ts [hunk 1]
```diff
@@ -75,6 +75,11 @@ describe("extractImageDetails", () => {
       imageName: "foo/bar",
       tag: "sha256:8756a25c4c5e902c4fe20322cc69d510a0517b51eab630c614efbd612ed568bf",
     }}
+    ${"localhost:1234/foo/bar:alpine3.15@sha256:8756a25c4c5e902c4fe20322cc69d510a0517b51eab630c614efbd612ed568bf"} | ${{
+      hostname: "localhost:1234",
+      imageName: "foo/bar",
+      tag: "sha256:8756a25c4c5e902c4fe20322cc69d510a0517b51eab630c614efbd612ed568bf",
```
Comment (gergo-papp) [inline test/lib/analyzer/image-inspector.spec.ts:81]: comment for a future discussion:

technically this is not a `tag` - it's a `digest`. should we model this differently in the future?

**Review by gergo-papp — APPROVED** [review]

File: test/lib/analyzer/image-inspector.spec.ts (hunk 1 above)
Comment (rmutuleanu) [reply test/lib/analyzer/image-inspector.spec.ts:81]: Yes, that is right. I think the terminology is not consistent everywhere.

**Review by RotemS — APPROVED** [review]
LGTM

**Pull request merged**